crates/services/p2p/src/p2p_service.rs

```diff
@@ -311,21 +344,53 @@ impl<Codec: NetworkCodec> FuelP2PService<Codec> {
         Ok(())
     }
 
+    pub fn update_block_height(&mut self, block_height: BlockHeight) {
+        self.swarm.behaviour_mut().update_block_height(block_height)
+    }
+
+    /// The report is forwarded to gossipsub behaviour
+    /// If acceptance is "Rejected" the gossipsub peer score is calculated
+    /// And if it's below allowed threshold the peer is banned
     pub fn report_message_validation_result(
         &mut self,
         msg_id: &MessageId,
-        propagation_source: &PeerId,
+        propagation_source: PeerId,
         acceptance: MessageAcceptance,
-    ) -> Result<bool, PublishError> {
-        self.swarm.behaviour_mut().report_message_validation_result(
-            msg_id,
-            propagation_source,
-            acceptance,
-        )
+    ) {
+        if let Some(gossip_score) = self
+            .swarm
+            .behaviour_mut()
+            .report_message_validation_result(msg_id, &propagation_source, acceptance)
+        {
+            let min_score_allowed =
+                self.network_metadata.gossipsub_data.min_score_allowed;
+
+            if gossip_score < min_score_allowed
+                && !self.peer_manager.is_reserved_peer(&propagation_source)
+            {
+                self.swarm.ban_peer_id(propagation_source);
+            }
+        }
     }
 
-    pub fn update_block_height(&mut self, block_height: BlockHeight) {
-        self.swarm.behaviour_mut().update_block_height(block_height)
+    /// Report application score
+    /// If application peer score is below allowed threshold
+    /// the peer is banend
+    pub fn report_peer(
+        &mut self,
+        peer_id: PeerId,
+        app_score: AppScore,
+        reporting_service: &str,
+    ) {
+        if let Some(PeerScoreUpdated { should_ban, score }) =
+            self.peer_manager.update_app_score(peer_id, app_score)
```

Comment:
I think we don't need to hesitate and don't pass the `&mut self.swarm` into `update_app_score`. If you pass it there, you can remove `PeerScoreUpdated` and simplify the logic. It is also applyable to other places that return `should_ban`

Reply:
resolved with #1118